R/plotGMA.R

```diff
@@ -0,0 +1,270 @@
+#' modes: split, complete and genelist.
+#' @param DEG_Methylation_Annotations A tibble which is outputted
+#' from the GMA function.
+#' @param Oncogenic_mediators_methylation_summary A tibble which
+#' is outputted from the GMA function.
+#' @param type A character string which can either be split,
+#' complete or genelist. If type is set to split, the entire
+#' dataset is split into groups of 40 genes and individual
+#' heatmaps of groups each containing 40 genes will be created and
+#' subsequently merged into one pdf where each page in the pdf
+#' is an individual heatmap. The genes will be sorted alphabetically. 
+#' If type is set to complete, a single heatmap is created where
+#' the number of differentially methylated CpG sites are shown for
+#' all oncogenic mediators. If type is set to genelist, a single
+#' heatmap will be created for genes supplied by the user in the
+#' genelist parameter. Default is split. 
+#' @param genelist A character string containing HUGO symbols
+#' of genes to be visualized in a single heatmap. Default is NULL.
+#' @param additionalFilename A character string that can be used
+#' to add a prefix or filepath to the filename of the pdf 
+#' visualizing the heatmap. Default is an empty string. 
+#' @import dplyr
+#' @import tidyr
+#' @import ComplexHeatmap
+#' @import tidyHeatmap
+#' @importFrom qpdf pdf_combine
+#' @importFrom purrr map
+#' @importFrom magrittr "%>%"
+#' @return No value is returned. Visualizations in form of heatmaps
+#' are saved.
+#' @export
+#' @examples 
+#' data("DEG_Methylation_Annotations")
+#' data("Oncogenic_mediators_methylation_summary")
+#' genes <- c("ACAN", "ACE2", "ADAM19", "AFAP1L1")
+#' plotGMA(DEG_Methylation_Annotations = DEG_Methylation_Annotations, 
+#'         Oncogenic_mediators_methylation_summary = Oncogenic_mediators_methylation_summary, 
+#'         genelist = genes, 
+#'         additionalFilename = "./GMAresults/")
+plotGMA <- function(DEG_Methylation_Annotations,
+                    Oncogenic_mediators_methylation_summary,
+                    type = "split",
+                    genelist = NULL,
+                    additionalFilename = "") {
+  
+  # Get logFC of DEGs
+  DEGs_logFC <- DEG_Methylation_Annotations %>% 
+    dplyr::select(Hugo_Symbol, 
+                  logFC) %>% 
+    distinct()
+  
+  # Reformat Oncogenic_mediators_methylation_summary data to long format
+  # needed for visualization in heatmap
+  summary_heatmap <- Oncogenic_mediators_methylation_summary %>% 
+    pivot_longer(data = ., 
+                 cols = contains("methyl_num"),
+                 names_to = "Met_type",
+                 values_to = "Met_count") %>% 
+    dplyr::select(-contains("NCG")) %>% 
+    replace_na(list(Met_count = 0)) %>% 
+    left_join(x = .,
+              y = DEGs_logFC,
+              by = "Hugo_Symbol")
+  
+  ## Genelist mode
+  
+  # If genelist is supplied
+  if (length(genelist) > 0) {
```

Comment:
here we should just check that the `type` is `genelist` instead (which we don't otherwise)

Reply:
updated